client/components/SendZoopDialog.tsx

```diff
@@ -0,0 +1,113 @@
+import SendIcon from "@mui/icons-material/Send"
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import TextField from "@mui/material/TextField";
+import React, { useState } from "react";
+
+import { 
+    useGetAllUsersQuery, 
+    useGetMeQuery, 
+    useSendZoopMutation 
+} from "../features/api";
+import { Zoop } from "../../src/types/custom";
+import ZoopSentDialog from "./ZoopSentDialog";
+
+export interface SendZoopDialogProps {
+    open: boolean
+    onClose: () => void
+}
+
+const SendZoopDialog = ({open, onClose}: SendZoopDialogProps) => {
+    const [username, setUsername] = useState("");
+    const [content, setContent] = useState("");
+
+    const currentUser = useGetMeQuery().data;
+
+    const [sendZoop, { isLoading: isSendZoopLoading, isError, data: zoopData, error: zoopError }] = useSendZoopMutation();
+    const {data: usersData, isLoading: isGetAllUsersLoading, error: userError } = useGetAllUsersQuery(); 
+
+    if (userError) {
+        console.error(userError);
+    }
+
+    if (zoopError) {
+        console.error(zoopError);
+    }
+
+    if (zoopData) {
+        console.log(zoopData);
+    }
+
+    const handleSendZoopClick = () => {
+        if (!isGetAllUsersLoading && !usersData) {
+            throw new Error('No users to send Zoop to');
+        } else if (!isGetAllUsersLoading && usersData) {
+            const [recipient] = usersData.users.filter(el => el.user.username === username);
+            if (recipient && currentUser) {
+                sendZoop({
+                    content: content,
+                    authorId: currentUser.user.id,
+                    receiverId: recipient.user.id
+                })
+            } else {
+                throw new Error('No such user exists. Please select a different username');
+            }  
+        }  
+    }
+
+    if (zoopData) {
+        return (
+            <ZoopSentDialog
+                sentZoop={zoopData.zoop}
+                onClose={onClose}
+            />
+        )
+    }
+    return (
+        <Dialog
+            open={open}
+            onClose={onClose}
+        >
+            <DialogTitle>Send Zoop</DialogTitle>
+            <DialogContent>
+                <TextField
```

Comment:
I could swear I left a comment here suggesting you take a look at the `Autocomplete` component in MUI instead of just using a text input. That would allow them to start typing a username and then see a filtered list of all usernames that match

Reply:
Yeah, I got that comment. It's done!